server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteController.java

```diff
@@ -45,61 +55,119 @@ public CheckinNoteController(CheckinNoteServices checkinNoteServices) {
                 .body(error);
     }
 
+    @Error(exception = CheckinNotesNotFoundException.class)
+    public HttpResponse<?> handleNotFound(HttpRequest<?> request, CheckinNotesNotFoundException e) {
+        JsonError error = new JsonError(e.getMessage())
+                .link(Link.SELF, Link.of(request.getUri()));
+
+        return HttpResponse.<JsonError>notFound()
+                .body(error);
+    }
+
+    @Error(exception = CheckinNotesBulkLoadException.class)
+    public HttpResponse<?> handleBulkLoadException(HttpRequest<?> request, CheckinNotesBulkLoadException e) {
+        return HttpResponse.badRequest(e.getErrors())
+                .headers(headers -> headers.location(request.getUri()));
+    }
+
+    @Error(exception = CompositeException.class)
+    public HttpResponse<?> handleRxException(HttpRequest<?> request, CompositeException e) {
+
+        for (Throwable t : e.getExceptions()) {
+            if (t instanceof CheckinNotesBadArgException) {
+                return handleBadArgs(request, (CheckinNotesBadArgException) t);
+            }
+            else if (t instanceof CheckinNotesNotFoundException) {
+                return handleNotFound(request, (CheckinNotesNotFoundException) t);
+            }
+        }
+
+        return HttpResponse.<JsonError>serverError();
+    }
+
     /**
      * Create and Save a new check in note
      *
-     * @param checkinNote
+     * @param checkinNote, {@link CheckinNoteCreateDTO}
      * @param request
-     * @return
+     * @return {@link HttpResponse <CheckinNote>}
      */
     @Post("/")
     @Secured({RoleType.Constants.PDL_ROLE, RoleType.Constants.ADMIN_ROLE})
-    public HttpResponse<CheckinNote> createCheckinNote(@Body @Valid CheckinNoteCreateDTO checkinNote, HttpRequest<CheckinNoteCreateDTO> request) {
-        CheckinNote newCheckinNote = checkinNoteServices.save(new CheckinNote(checkinNote.getCheckinid(), checkinNote.getCreatedbyid()
-                , checkinNote.getDescription()));
-        return HttpResponse.created(newCheckinNote)
-                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getPath(), newCheckinNote.getId()))));
-
+    public Single<HttpResponse<CheckinNote>> createCheckinNote(@Body @Valid CheckinNoteCreateDTO checkinNote,
+                                                                    HttpRequest<CheckinNoteCreateDTO> request) {
+        return Single.fromCallable(() -> checkinNoteServices.save(new CheckinNote(checkinNote.getCheckinid(),
+                checkinNote.getCreatedbyid(), checkinNote.getDescription())))
+                .observeOn(Schedulers.from(eventLoopGroup))
+                .map(createdCheckinNote -> {
+                    //Using code block rather than lambda so we can log what thread we're in
```

Comment:
This comment was here for demonstrative purposes. You can get rid of it and change this to a lambda and the result will be a much clearer read.

Reply:
Updated